node/src/lib.rs

```diff
@@ -132,7 +132,15 @@ impl Node {
         let initial_committee = ReconfigureNotification::NewEpoch((**committee.load()).clone());
         let (tx_reconfigure, _rx_reconfigure) = watch::channel(initial_committee);
 
-        let (tx_new_certificates, rx_new_certificates) = channel(Self::CHANNEL_CAPACITY);
+        // This gauge is porcelain: do not modify it without also modifying `primary::metrics::PrimaryChannelMetrics::replace_registered_new_certificates_metric`
+        // This hack avoids a cyclic dependency in the initialization of consensus and primary
+        let new_certificates_counter = IntGauge::new(
+            "tx_new_certificates",
+            "occupancy of the channel from the `primary::Core` to the `Consensus`",
+        )
+        .unwrap();
+        let (tx_new_certificates, rx_new_certificates) =
+            types::metered_channel::channel(Self::CHANNEL_CAPACITY, &new_certificates_counter);
```

Comment:
Here's the issue: even if I introduce it as consensus metrics, we cannot make this part of the initialization of consensus, because we would need to:
- create the channel in the node, with its associated metric
- pass the sending end to the Consensus, and probably re-register the associated metric there,
- and then pass the receiving end of the channel to the primary.

This is the exact symmetric of what we're doing now:
- create the channel in the node initialization, with its associated metric
- pass the sending end to the consensus,
- and then pass the receiving end of the channel to the primary, re-register the associated metric there,

I'm happy to let you have a go at it in a follow-up PR, if you have an idea of how to clean this up.